bitrise.yml

```diff
@@ -0,0 +1,138 @@
+
+            if [[ "${BITRISE_GIT_BRANCH}" != "master" ]]; then
+              echo " [!] Deployments are allowed only on 'master' branch"
+              exit 1
+            fi
+    - script@1:
+        title: Lint and push Cocoapods trunk
+        inputs:
+        - content: |-
+            #!/usr/bin/env bash
+            set -ex
+
+            bundle exec pod spec lint --allow-warnings
+            bundle exec pod trunk push --allow-warnings --verbose
+    - script@1:
+        title: Generate docs
+        inputs:
+        - content: |-
+            #!/usr/bin/env bash
+            set -ex
+
+            bundle exec jazzy
+    - script@1:
+        title: Deploy Docs to Github Pages
+        inputs:
+        - content: |-
+            #!/usr/bin/env bash
+            set -ex
+
+            # Remove docs from gitignore
+            rm .gitignore
+            git add docs
+            git commit -m "Deploy Jazzy docs for version $BITRISE_GIT_TAG"
+
+            git config --global url."https://x-token-auth:$PUBLISHER_GITHUB_API_TOKEN@github.com".InsteadOf https://github.com
+            git subtree push --prefix docs origin gh-pages
+            git config --global --unset url."https://github.com".insteadOf
+
+  _setup:
+    steps:
+    - git-clone@4:
+        inputs:
+        - update_submodules: 'no'
+    - cache-pull@2: {}
+    - script@1:
+        title: Bundle install
+        inputs:
+        - content: |-
+            #!/usr/bin/env bash
+            set -ex
+            if [ ! -f "Gemfile" ]; then
+                echo "No Gemfile detected. Skipping..."
+                exit 0
+            fi
+            bundle install
+            RBENV_DIR="`cd $(rbenv which ruby)/../..;pwd`"
+            echo "Gem cache directory: $RBENV_DIR"
+            envman add --key GEM_CACHE_PATH --value $RBENV_DIR
+    - brew-install@0:
+        title: Install swiftlint
+        inputs:
+        - cache_enabled: 'yes'
+        - upgrade: 'no'
+        - packages: swiftlint
+    - cocoapods-install@1:
+        inputs:
+        - verbose: 'false'
+
+  _report:
+    steps:
+    - danger@2:
+        inputs:
+        - github_api_token: "$DANGER_GITHUB_API_TOKEN"
+    - cache-push@2:
+        inputs:
+        - cache_paths: |-
+            $BITRISE_CACHE_DIR
+            $GEM_CACHE_PATH
+    - script@1:
+        title: Send XCov report
+        inputs:
+        - content: |-
+            #!/usr/bin/env bash
+            set -ex
+
+            bash <(curl -s https://codecov.io/bash) -J '^RInAppMessaging$' -X gcov
+app:
+  envs:
+  - opts:
+      is_expand: false
+    BITRISE_PROJECT_PATH: RInAppMessaging.xcworkspace
+  - opts:
+      is_expand: false
+    BITRISE_SCHEME: RInAppMessaging-Example
+  - opts:
+      is_expand: false
+    BITRISE_EXPORT_METHOD: ad-hoc
+  - opts:
+      is_expand: false
+    CODECOV_TOKEN: 4144ada4-65b7-44bd-b478-48e3583a55cd
+  - opts:
+      is_expand: false
+    DANGER_GITHUB_API_TOKEN: 78fb7ddf8a020d492ed125f3277d631b3a5b7e6c
```

Comment:
GitHub revoked this token due to being included in this commit.
It will need to be injected as an env var from the bitrise web config. I generated a new token and added it to the accounts page.

Reply:
Oh noo... :( This mean we can't use repo based bitrise.yml. The best we can do is to use that workaround we have in our other projects - that way we'll be able to keep env vars on the web and workflows in the repo.